ui/app/services/path-help.js

```diff
@@ -42,89 +44,126 @@ export default Service.extend({
       if (newModel.merged || modelProto.useOpenAPI !== true) {
         return resolve();
       }
+
       helpUrl = modelProto.getHelpUrl(backend);
       return this.registerNewModelWithProps(helpUrl, backend, newModel, modelName);
     } else {
       debug(`Creating new Model for ${modelType}`);
       newModel = DS.Model.extend({});
-      //use paths to dynamically create our openapi help url
-      //if we have a brand new model
-      return this.getPaths(apiPath, backend, itemType).then(paths => {
+    }
+
+    //we don't have an apiPath for dynamic secrets
+    //and we don't need paths for them yet
+    if (!apiPath) {
+      helpUrl = newModel.proto().getHelpUrl(backend);
+      return this.registerNewModelWithProps(helpUrl, backend, newModel, modelName);
+    }
+
+    //use paths to dynamically create our openapi help url
+    //if we have a brand new model
+    return this.getPaths(apiPath, backend, itemType)
+      .then(pathInfo => {
         const adapterFactory = owner.factoryFor(`adapter:${modelType}`);
         //if we have an adapter already use that, otherwise create one
         if (!adapterFactory) {
           debug(`Creating new adapter for ${modelType}`);
-          const adapter = this.getNewAdapter(paths, itemType);
+          const adapter = this.getNewAdapter(pathInfo, itemType);
           owner.register(`adapter:${modelType}`, adapter);
         }
+        let path, paths;
         //if we have an item we want the create info for that itemType
-        let path;
-        if (itemType) {
-          const createPath = paths.create.find(path => path.path.includes(itemType));
-          path = createPath.path;
-          path = path.slice(0, path.indexOf('{') - 1) + '/example';
-        } else {
-          //we need the mount config
-          path = paths.configPath[0].path;
+        paths = itemType ? this.filterPathsByItemType(pathInfo, itemType) : pathInfo.paths;
+        const createPath = paths.find(path => path.operations.includes('post') && path.action !== 'Delete');
+        path = createPath.path;
+        path = path.includes('{') ? path.slice(0, path.indexOf('{') - 1) + '/example' : path;
+        if (!path) {
+          return reject();
```

Comment:
I think maybe adding a note here about why we `reject` with the `catch` below, instead of just `resolve` here. I'm guess it's so we don't have to guard for no args anytime `getPaths` is used, but it was a bit of a surprise to see it. Do we know how common it is that there won't be a path here?  It looks like if there's no `createPath` we never register the model - I think that's worth commenting too as it may be something that we'd want to change later.

Reply:
good question! i'm not entirely sure if `path` is ever false-y, so i've added a todo comment here to come back to it.